oak/server/rust/oak_runtime/src/metrics/server.rs

```diff
@@ -53,25 +54,47 @@ async fn serve_metrics(_req: Request<Body>) -> Result<Response<Body>, MetricsSer
     info!("Metrics size: {}", buffer.len());
 
     Response::builder()
-        .status(http::StatusCode::OK)
+        .status(StatusCode::OK)
         .header(CONTENT_TYPE, encoder.format_type())
         .body(Body::from(buffer))
         .map_err(|e| {
             MetricsServerError::ResponseError(format!("Could not build the response: {}", e))
         })
 }
 
-async fn make_server(port: u16, notify_receiver: tokio::sync::oneshot::Receiver<()>) {
+async fn serve_metrics(
+    runtime: Arc<Runtime>,
+    req: Request<Body>,
+) -> Result<Response<Body>, MetricsServerError> {
+    match (req.method(), req.uri().path()) {
+        (&Method::GET, "/metrics") => handle_metrics_request(&runtime).await,
+        _ => Ok(Response::builder()
+            .status(StatusCode::NOT_FOUND)
+            .body(Body::from("Not Found!\n"))
+            .unwrap()),
+    }
+}
+
+async fn make_server(
+    port: u16,
+    runtime: Arc<Runtime>,
+    notify_receiver: tokio::sync::oneshot::Receiver<()>,
+) {
     let addr = SocketAddr::from(([0, 0, 0, 0], port));
 
     // A `Service` is needed for every connection, so this
     // creates one from the `process_metrics` function.
-    let make_svc = make_service_fn(|_conn| async {
-        // service_fn converts our function into a `Service`
-        Ok::<_, hyper::Error>(service_fn(serve_metrics))
+    let make_service = make_service_fn(move |_conn| {
+        let runtime = runtime.clone();
+        async move {
+            Ok::<_, hyper::Error>(service_fn(move |req| {
+                let runtime = runtime.clone();
+                async move { serve_metrics(runtime, req).await }
```

Comment:
I would have thought that `async { foo.await }` is equivalent to just `foo`, or is this closure needed because of the explicit move?

Reply:
Yes. Thanks. Fixed. Must have been leftover from my chain of changes to make Rust happy.